src/utils/json.ts

```diff
@@ -0,0 +1,120 @@
+// Regex to clean up added $refId & $ref properties, added by Realm.JsonSerializationReplacer
+
+import { display as displayDataCell } from '../ui/RealmBrowser/Content/Table/types/DataCell';
+
+// TODO: Investigate better solution.
+const $REF_MATCHER = /\s*\"\$ref[Id]*\" *: *(\"(.*?)\"(,|\s|)|\s*\{(.*?)\}(,|\s|))/g;
+
+const DEFAULT_POSTFIX = ' (...)';
+const INDENTATION = 2;
+
+type SafeJsonOptions = {
+  pretty?: boolean;
+  cleanupRefs?: boolean;
+  maxLength?: number;
+  postFix?: string;
+};
+
+export const asSafeJsonString = (
+  value: unknown,
+  options?: SafeJsonOptions,
+): string => {
+  let json: string;
+  const indentation = options?.pretty ? INDENTATION : undefined;
+
+  try {
+    json = JSON.stringify(value, null, indentation);
+  } catch {
+    try {
+      json = JSON.stringify(
+        value,
+        Realm.JsonSerializationReplacer,
+        indentation,
+      );
+    } catch (err) {
+      json = err.message ? err.message : err;
+    }
+  }
+
+  if (!json) {
+    return 'null';
+  }
+
+  if (options) {
+    if (options.cleanupRefs) {
+      json = json.replace($REF_MATCHER, '');
+    }
+
+    if (options.maxLength && json.length > options.maxLength) {
+      json =
+        json.slice(0, options.maxLength) + (options.postFix ?? DEFAULT_POSTFIX);
+    }
+  }
+
+  return json;
+};
+
+/**
+ * Utility function to indicate if the JsonViewerDialog should be utilized for a cell
+ */
+export const useJsonViewer = (
```

Comment:
```suggestion
export const needsJsonViewer = (
```

The `use` prefix makes me think this is a hook.

Reply:
I went with a compromise: `canUseJsonViewer`.
The intent of the function is just to have the logic in _one_ place - a goal I would assume would shift as the JsonViewer is no longer needed, or extended.